### ISSUE TITLE
Java coverage does not work

### ISSUE THREAD

**Issue by lberki** [bazelbuild/bazel#2820]
Java coverage does not work. Furthermore, its test have been disabled in bce889840f91f14c05418d960984bcf1a5f59411 by @katre .

The erorr message is:

```
Exception in thread "Thread-0" java.lang.NoClassDefFoundError: org/objectweb/asm/ClassVisitor
	at com.google.testing.coverage.JacocoCoverageRunner.analyzeBranch(JacocoCoverageRunner.java:138)
	at com.google.testing.coverage.JacocoCoverageRunner.create(JacocoCoverageRunner.java:84)
	at com.google.testing.coverage.JacocoCoverageRunner$2.run(JacocoCoverageRunner.java:258)
Caused by: java.lang.ClassNotFoundException: org.objectweb.asm.ClassVisitor
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:332)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
```

I checked `JacocoCoverageRunner_deploy.jar` and it's consistent with the message above: `ClassVisitor` is renamed by the `jarjar` invocation, but the reference to it in `BranchDetailAnalyzer` is not. Digging.

**Comment by damienmg** [comment]
The test got reactivated at https://github.com/bazelbuild/bazel/commit/21fb8cac60c36cb527d6131ab0092bcdf138d99a

**Comment by damienmg** [comment]
Maybe we should use coverage on Bazel itself, that would be nice and would give more coverage for testing coverage :P

**Comment by lberki** [comment]
https://github.com/bazelbuild/bazel/blob/master/src/test/shell/bazel/bazel_coverage_test.sh#L111 says they haven't been reactivated.

**Comment by lberki** [comment]
Progress: this is because

1. The `jarjar` version we have (1.4) silently refuses to process class files with major version > 51 (Java 7)
2. We build the coverage test runner with `-target 8` by default

Confirmed by trying to rename some classes with `jarjar` in a little test project, observing that it silently does nothing, rebuilding said project with `-target 7 -source 7` and observing that then `jarjar` works.

**Comment by lberki** [comment]
This was fixed in 3a5d3db6436bbaf5351da9ea9c19a41ae7c95ca9 .
